src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticListener.cs

```diff
@@ -63,7 +63,7 @@ public static IObservable<DiagnosticListener> AllListeners
         ///
         /// If this parameter is null, no filtering is done (all overloads of DiagnosticSource.IsEnabled return true).
         /// </param>
-        public virtual IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
+        public virtual IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string>? isEnabled)
```

Comment:
I don't see anything from preventing the observer have `null` strings passed down.

Reply:
It's referenced without null check https://github.com/dotnet/corefx/pull/41241/files/1daae397b0214c469095d74cc550a7946307d5b2#diff-c1b7bd184c6f375db1ed7298b13a29feR262